modules/docker/inspect.go

```diff
@@ -0,0 +1,204 @@
+	}
+	HostConfig struct {
+		Binds []string
+	}
+}
+
+func Inspect(t *testing.T, id string) ContainerInspect {
+	out, err := InspectE(t, id)
+	require.NoError(t, err)
+	return out
+}
+
+// InspectE runs the 'docker inspect {container id}' command and returns a ContainerInspect
+// struct, converted from the output JSON, along with any errors
+func InspectE(t *testing.T, id string) (ContainerInspect, error) {
+	match, _ := regexp.MatchString("[[:alnum:]]", id)
+	if !match {
+		return ContainerInspect{}, fmt.Errorf("%s is not a valid container ID", id)
+	}
+
+	cmd := shell.Command{
+		Command: "docker",
+		Args:    []string{"container", "inspect", id},
+	}
+
+	out, err := shell.RunCommandAndGetOutputE(t, cmd)
+	if err != nil {
+		return ContainerInspect{}, err
+	}
+
+	var containers []inspectOutput
+	err = json.Unmarshal([]byte(out), &containers)
+	if err != nil {
+		return ContainerInspect{}, err
+	}
+
+	if len(containers) == 0 {
+		return ContainerInspect{}, nil
+	}
+
+	c := containers[0]
+
+	return transformContainer(t, c), nil
+}
+
+// transformContainerPorts converts 'docker inspect' output JSON into a more friendly and testable format
+func transformContainer(t *testing.T, c inspectOutput) ContainerInspect {
+	name := strings.TrimLeft(c.Name, "/")
+	ports := transformContainerPorts(t, c)
+	volumes := transformContainerVolumes(t, c)
+
+	created, err := time.Parse(time.RFC3339Nano, c.Created)
+	require.NoError(t, err)
+
+	inspect := ContainerInspect{
+		ID:       c.Id,
+		Name:     name,
+		Created:  created,
+		Status:   c.State.Status,
+		Running:  c.State.Running,
+		ExitCode: c.State.ExitCode,
+		Error:    c.State.Error,
+		Ports:    ports,
+		Binds:    volumes,
+	}
+
+	return inspect
+}
+
+// transformContainerPorts converts Docker's ports from the following json into a more testable format
+// {
+//   "80/tcp": [
+//     {
+// 	     "HostIp": ""
+//       "HostPort": "8080"
+//     }
+//   ]
+// }
+func transformContainerPorts(t *testing.T, c inspectOutput) []Port {
+	var ports []Port
+
+	cPorts := c.NetworkSettings.Ports
+
+	for k, pb := range cPorts {
+		split := strings.Split(k, "/")
+
+		containerPort, err := strconv.ParseUint(split[0], 10, 16)
+		require.NoError(t, err)
+
+		protocol := split[1]
+
+		for _, p := range pb {
+			hostPort, err := strconv.ParseUint(p.HostPort, 10, 16)
+			require.NoError(t, err)
+
+			ports = append(ports, Port{
+				HostPort:      uint16(hostPort),
+				ContainerPort: uint16(containerPort),
+				Protocol:      protocol,
+			})
+		}
+	}
+
+	return ports
+}
+
+// transformContainerVolumes converts Docker's volume bindings from the
+// format "/foo/bar:/foo/baz" into a more testable one
+func transformContainerVolumes(t *testing.T, c inspectOutput) []VolumeBind {
+	binds := c.HostConfig.Binds
+	volumes := make([]VolumeBind, 0, len(binds))
+
+	for _, b := range binds {
+		var source, dest string
+
+		split := strings.Split(b, ":")
+
+		// Considering it a named volume
+		source = split[0]
+		dest = split[1]
```

Comment:
Check `len(split) == 2`!

Reply:
From what I have tested Docker won't return a HostConfig.Binds without a colon, but I'm not 100% sure on the internals so it makes sense to add a check just to be safe.

Also gonna add tests for named and anonymous volumes, sounds safer that way.